modules/afsocket/socket-options-inet.c

```diff
@@ -77,6 +94,21 @@ socket_options_inet_setup_socket(SocketOptions *s, gint fd, GSockAddr *addr, AFS
 #endif
     }
 
+  if (self->interface_name)
+    {
+#ifdef SO_BINDTODEVICE
+      int result = socket_options_inet_set_interface(self, fd);
+      if (result != 0)
+        {
+          msg_error("Can't bind to interface", evt_tag_str("interface", self->interface_name), evt_tag_errno("error", result));
+          return FALSE;
+        }
+#else
+      msg_error("interfacel() is set but no SO_BINDTODEVICE setsockopt on this platform");
+      return FALSE;
+#endif
```

Comment:
Optional review note:

This function is already huge. Could you extract this code with those ifdefs, for example:

```
if (self->interface_name && !socket_options_inet_try_bind_to_interface(self, fd))
  return FALSE;
```

or

```
if (self->interface_name)
{
  if (!socket_options_inet_try_bind_to_interface(self, fd))
    return FALSE;
}
```